### PR TITLE
Readme

### PR DESCRIPTION
**Pull request by nawarani**
Added MVC diagram

### CODE REVIEW

**Review by AscendedWise — APPROVED** [review]
The readme looks fantastic

**Comment by 123Chan** [comment]
It looks good, but some things should be changed:

- View doesn't depend on Presenter, it only depends on Controller and implements View Interface
- I think you should add method descriptions in presenter interface and view interface, similar to how you did for Use Cases

File: README.md [hunk 1]
```diff
@@ -29,6 +28,9 @@ It features all common chess mechanisms, including move generation, move validat
 - Architecture Pattern: Model-View-Controller (MVC)
 - Board Representation: Bitboard
 
+This project uses MVC pattern to deploy clean architecture. While some interdependency is unavoidable due to the nature of chess as a game, all such dependencies are contained within each layer and adhere to SOLID principles.
+![MVC](https://i.imgur.com/wci7Tt4.png)
```
Comment (123Chan) [inline README.md:32]: This image should be changed. See comment.

**Review by 123Chan — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (nawarani) [reply README.md]: Changed

**Review by AscendedWise — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (123Chan) [reply README.md]: Looks great. The diagram demonstrates our MVC-Clean Architecture implementation.

**Review by 123Chan — APPROVED** [review]

**Pull request merged**